tests/test_all_component_and_model_builder.py

```diff
@@ -0,0 +1,296 @@
+    )
+
+
+def test_subset_component(sub_set_model):
+    """Test if model builder detects the subset component.
+    """
+    mb = ModelBuilder()
+    # Test subset
+    superset = mb._is_subset_component(sub_set_model)
+    assert superset == "Spindown"
+
+
+def test_model_from_par():
+    """Test Get model from test par file.
+    """
+    test_par1 = """
+    PSR              B1855+09
+    LAMBDA   286.8634893301156  1     0.0000000165859
+    BETA      32.3214877555037  1     0.0000000273526
+    PMLAMBDA           -3.2701  1              0.0141
+    PMBETA             -5.0982  1              0.0291
+    PX                  0.2929  1              0.2186
+    F0    186.4940812707752116  1  0.0000000000328468
+    F1     -6.205147513395D-16  1  1.379566413719D-19
+    PEPOCH        54978.000000
+    START            53358.726
+    FINISH           56598.873
+    DM               13.299393
+    DMX              14.000000
+    DMX_0001    1.51618630D-02  1      3.51684846D-03
+    DMXEP_0001     53358.72746
+    DMXR1_0001     53358.72746
+    DMXR2_0001     53358.77841
+    DMXF1_0001         424.000
+    DMXF2_0001        1442.000
+    DMX_0002    1.52370685D-02  1      3.51683449D-03
+    DMXEP_0002     53420.54893
+    DMXR1_0002     53420.54893
+    DMXR2_0002     53420.58620
+    DMXF1_0002         424.000
+    DMXF2_0002        1442.000
+    FD1  1.61666384D-04  1      3.38650356D-05
+    FD2 -1.88210030D-04  1      4.13173074D-05
+    SOLARN0               0.00
+    EPHEM               DE421
+    ECL                 IERS2003
+    CLK                 TT(BIPM)
+    UNITS               TDB
+    TIMEEPH             FB90
+    T2CMETHOD           TEMPO
+    CORRECT_TROPOSPHERE N
+    PLANET_SHAPIRO      N
+    #DILATEFREQ          N
+    NTOA                  4005
+    TRES                  5.52
+    TZRMJD  54981.28084616488447
+    TZRFRQ             424.000
+    TZRSITE                  3
+    NITS                     1
+    INFO -f
+    BINARY            DD
+    A1             9.230780480  1         0.000000203
+    E             0.0000216340  1        0.0000000236
+    T0        54975.5128660817  1        0.0019286695
+    PB       12.32717119132762  1    0.00000000019722
+    OM        276.536118059963  1      0.056323656112
+    SINI              0.999461  1            0.000178
+    M2                0.233837  1            0.011278
+    RNAMP         0.17173D-01
+    RNIDX            -4.91353
+    TNRedAmp -14.227505410948254
+    TNRedGam 4.91353
+    TNRedC 45
+    T2EFAC -f L-wide_PUPPI   1.507
+    T2EFAC -f 430_ASP   1.147
+    T2EQUAD -f L-wide_PUPPI   0.25518
+    T2EQUAD -f 430_ASP   0.01410
+    ECORR -f 430_PUPPI   0.00601
+    ECORR -f L-wide_PUPPI   0.31843
+    JUMP -fe L-wide      -0.000009449  1       0.000009439
+    """
+    mb = ModelBuilder()
+    param_inpar, repeat = mb.parse_parfile(io.StringIO(test_par1))
+    assert len(param_inpar) == 60
+    assert len(repeat) == 4
```

Comment:
Please test what repeat is more explicitly - if I run the test suite and find out it's not 4 I won't have any idea why it should be 4. I think here it should be the set `{JUMP, T2EQUAD, T2EFAC, ECORR}`?